src/test/java/uk/gov/hmcts/juror/api/moj/report/grouped/TrialAttendanceReportTest.java

```diff
@@ -0,0 +1,190 @@
+            .reportType(report.getName())
+            .trialNumber("TRIALNUMBER")
+            .build();
+    }
+
+    @Override
+    public void positivePreProcessQueryTypical(JPAQuery<Tuple> query, StandardReportRequest request) {
+        String locCode = "415";
+        TestUtils.mockSecurityUtil(BureauJwtPayload.builder().locCode(locCode).userType(UserType.COURT).build());
+
+        report.preProcessQuery(query, request);
+        verify(query, times(1))
+            .where(QReportsJurorPayments.reportsJurorPayments.trialNumber.eq(request.getTrialNumber()));
+        verify(query, times(1))
+            .where(QReportsJurorPayments.reportsJurorPayments.locCode.eq(locCode));
+        verify(query, times(1)).orderBy(
+            QReportsJurorPayments.reportsJurorPayments.jurorNumber.asc());
+    }
+
+    @Override
+    public Map<String, GroupedReportResponse.DataTypeValue> positiveGetHeadingsTypical(
+        StandardReportRequest request,
+        GroupedReportResponse.TableData<GroupedTableData> tableData,
+        GroupedTableData data) {
+        TestUtils.mockSecurityUtil(BureauJwtPayload.builder()
+                                       .locCode(TestConstants.VALID_COURT_LOCATION)
+                                       .owner(TestConstants.VALID_COURT_LOCATION)
+                                       .userType(UserType.COURT)
+                                       .build());
+
+        String trialNumber = "TRIALNUMBER";
+        when(request.getTrialNumber()).thenReturn(trialNumber);
+
+        CourtLocation courtLocation = new CourtLocation();
+        courtLocation.setLocCode(TestConstants.VALID_COURT_LOCATION);
+        courtLocation.setName("CHESTER");
+        when(courtLocationRepository.findByLocCode(TestConstants.VALID_COURT_LOCATION))
+            .thenReturn(Optional.of(courtLocation));
+
+        Trial trial = Trial.builder()
+            .description("TRIAL_DEFENDANTS")
+            .trialType(TrialType.CRI)
+            .judge(Judge.builder()
+                       .name("JUDGE_NAME")
+                       .build())
+            .courtroom(Courtroom.builder()
+                           .description("ROOM_NAME")
+                           .build())
+            .courtLocation(courtLocation)
+            .trialStartDate(LocalDate.now().minusDays(1))
+            .build();
+        when(trialRepository.findByTrialNumberAndCourtLocationLocCode(request.getTrialNumber(),
+                                                                      TestConstants.VALID_COURT_LOCATION))
+            .thenReturn(Optional.of(trial));
+
+
+        Map<String, GroupedReportResponse.DataTypeValue> map = report.getHeadings(request, tableData);
+        assertHeadingContains(map, request, false, Map.of(
+            "trial_number", GroupedReportResponse.DataTypeValue.builder()
+                .displayName("Trial Number")
+                .dataType("String")
+                .value(request.getTrialNumber())
```

Comment:
Values should be hard coded to ensure it is what we are expecting (same for all)

Reply:
Fixed